### ISSUE TITLE
Help required to use new dataset

### ISSUE THREAD

**Issue by anshumanpatil** [tekaratzas/RustGPT#26]
In below PR - 

https://github.com/tekaratzas/RustGPT/pull/25

I have tried to use new dataset from hf got somewhat success but not completely successful

Completed : - 

**New dataset added related to fitness**

- Changed dataset formatting and created pre training data.
 
- Used dataset - [fitness-question-answers](https://huggingface.co/datasets/its-myrto/fitness-question-answers) 
 
- Created pre training data with - [ibm/granite-4-h-tiny](https://huggingface.co/ibm-granite/granite-4.0-h-tiny)
 
Tried but unsuccessful : -

**Working on** 

Adjusting hyperparameters.

- Changed `MAX_SEQ_LEN` and found that 180 works.

- Still working on `learning rate` and `batch size`

Any Suggestion/Help is welcome 
Thanks in advance

**Comment by tekaratzas** [comment]
What isn't working?

**Comment by anshumanpatil** [comment]
> What isn't working?

it hallucinates

basically gives weird answers

loss function do not reach below 0 at the end of training

I also tried learning rate and batch size

still working on it
